data/transactions/logic/assembler.go

```diff
@@ -0,0 +1,1053 @@
+		}
+		bvals = append(bvals, val)
+		rest = rest[consumed:]
+	}
+	var scratch [binary.MaxVarintLen64]byte
+	l := binary.PutUvarint(scratch[:], uint64(len(bvals)))
+	ops.Out.Write(scratch[:l])
+	for _, bv := range bvals {
+		l := binary.PutUvarint(scratch[:], uint64(len(bv)))
+		ops.Out.Write(scratch[:l])
+		ops.Out.Write(bv)
+	}
+	return nil
+}
+
+// addr A1EU...
+// parses base32-with-checksum account address strings into a byte literal
+func assembleAddr(ops *OpStream, args []string) error {
+	if len(args) != 1 {
+		return errors.New("addr operation needs one argument")
+	}
+	addr, err := basics.UnmarshalChecksumAddress(args[0])
+	if err != nil {
+		return err
+	}
+	return ops.ByteLiteral(addr[:])
+}
+
+func assembleArg(ops *OpStream, args []string) error {
+	val, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return err
+	}
+	return ops.Arg(val)
+}
+
+func assembleBnz(ops *OpStream, args []string) error {
+	if len(args) != 1 {
+		return errors.New("bnz operation needs label argument")
+	}
+	ops.ReferToLabel(ops.sourceLine, ops.Out.Len(), args[0])
+	ops.Out.WriteByte(0x40)
+	ops.Out.WriteByte(0)
+	ops.Out.WriteByte(0)
+	return nil
+}
+
+func assembleLoad(ops *OpStream, args []string) error {
+	val, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return err
+	}
+	if val > 255 {
+		return errors.New("load limited to 0..255")
+	}
+	ops.Out.WriteByte(0x34)
+	ops.Out.WriteByte(byte(val))
+	return nil
+}
+
+func assembleStore(ops *OpStream, args []string) error {
+	val, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return err
+	}
+	if val > 255 {
+		return errors.New("store limited to 0..255")
+	}
+	ops.Out.WriteByte(0x35)
+	ops.Out.WriteByte(byte(val))
+	return nil
+}
+
+// TxnFieldNames are arguments to the 'txn' and 'txnById' opcodes
+var TxnFieldNames = []string{
+	"Sender", "Fee", "FirstValid", "LastValid", "Note",
+	"Receiver", "Amount", "CloseRemainderTo", "VotePK", "SelectionPK",
+	"VoteFirst", "VoteLast", "VoteKeyDilution",
+	"Type", "TypeEnum",
+	"XferAsset", "AssetAmount",
+	"AssetSender", "AssetReceiver", "AssetCloseTo",
+	"GroupIndex", "TxID", "SenderBalance",
+}
+
+// TxnFieldTypes is StackBytes or StackUint64 parallel to TxnFieldNames
+var TxnFieldTypes = []StackType{
+	StackBytes, StackUint64, StackUint64, StackUint64, StackBytes,
+	StackBytes, StackUint64, StackBytes, StackBytes, StackBytes,
+	StackUint64, StackUint64, StackUint64,
+	StackBytes, StackUint64,
+	StackBytes, StackUint64,
+	StackBytes, StackBytes, StackBytes,
+	StackUint64, StackBytes, StackUint64,
+}
+
+var txnFields map[string]uint
+
+// TxnTypeNames is the values of Txn.Type in enum order
+var TxnTypeNames = []string{
+	string(protocol.UnknownTx),
+	string(protocol.PaymentTx),
+	string(protocol.KeyRegistrationTx),
+	string(protocol.AssetConfigTx),
+	string(protocol.AssetTransferTx),
+	string(protocol.AssetFreezeTx),
+}
+
+// map TxnTypeName to its enum index, for `txn TypeEnum`
+var txnTypeIndexes map[string]int
```

Comment:
Minor suggestion: move to `eval.go` because this is also used to implement `txn TypeEnum`.

Reply:
`TxnTypeNames` and `txnTypeIndexes` are _mostly_ used in assembler.go , and it's all the same package.